patches/net/minecraft/world/item/crafting/RecipeMap.java.patch

```diff
@@ -0,0 +1,36 @@
+--- a/net/minecraft/world/item/crafting/RecipeMap.java
++++ b/net/minecraft/world/item/crafting/RecipeMap.java
+@@ -13,7 +_,7 @@
+ 
+ public class RecipeMap {
+     public static final RecipeMap EMPTY = new RecipeMap(ImmutableMultimap.of(), Map.of());
+-    private final Multimap<RecipeType<?>, RecipeHolder<?>> byType;
++    private Multimap<RecipeType<?>, RecipeHolder<?>> byType;
+     private final Map<ResourceKey<Recipe<?>>, RecipeHolder<?>> byKey;
+ 
+     private RecipeMap(Multimap<RecipeType<?>, RecipeHolder<?>> p_379497_, Map<ResourceKey<Recipe<?>>, RecipeHolder<?>> p_380280_) {
+@@ -31,6 +_,24 @@
+         }
+ 
+         return new RecipeMap(builder.build(), builder1.build());
++    }
++
++    // Neo: handle ordering according to any recipe priorities
++    public void order(it.unimi.dsi.fastutil.objects.Object2IntMap<ResourceKey<net.minecraft.world.item.crafting.Recipe<?>>> recipePriorities) {
++        it.unimi.dsi.fastutil.ints.Int2ObjectMap<java.util.List<RecipeHolder<?>>> priorityBuilder = new it.unimi.dsi.fastutil.ints.Int2ObjectLinkedOpenHashMap<>();
```

Comment:
To order by priority (and not encounter order), I would use `Int2ObjectRBTreeMap`.